<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ImportDataDialog</class>
 <widget class="QDialog" name="ImportDataDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1039</width>
    <height>730</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Daten importieren</string>
  </property>
  <widget class="QGroupBox" name="grp_DataFile">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>561</width>
     <height>71</height>
    </rect>
   </property>
   <property name="title">
    <string>Datendatei</string>
   </property>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>30</y>
      <width>541</width>
      <height>33</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QLineEdit" name="edt_ImportFileName">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_BrowseCsv">
       <property name="text">
        <string>Datei öffnen</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="grp_AssignData">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>280</y>
     <width>1021</width>
     <height>271</height>
    </rect>
   </property>
   <property name="title">
    <string>Daten zuordnen</string>
   </property>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>442</x>
      <y>31</y>
      <width>258</width>
      <height>218</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QLabel" name="lbl_PartFields">
       <property name="text">
        <string>Part-Felder</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QListWidget" name="lst_PartFields">
       <property name="dragEnabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>31</y>
      <width>258</width>
      <height>218</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QLabel" name="lbl_SourceHeaders">
       <property name="text">
        <string>CSV-Spalten</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QListWidget" name="lst_CsvColumns">
       <property name="dragEnabled">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>731</x>
      <y>31</y>
      <width>258</width>
      <height>218</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_6">
     <item>
      <widget class="QLabel" name="lbl_PartFields_2">
       <property name="text">
        <string>Zuweisungs-Übersicht</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTableWidget" name="tbl_Mapping">
       <property name="editTriggers">
        <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
       </property>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>127</number>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderStretchLastSection">
        <bool>false</bool>
       </attribute>
       <column>
        <property name="text">
         <string>CSV-Spalte</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Part-Feld</string>
        </property>
       </column>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>30</y>
      <width>151</width>
      <height>221</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <spacer name="verticalSpacer_3">
       <property name="orientation">
        <enum>Qt::Orientation::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>10</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btn_MapToField">
       <property name="text">
        <string>zuweisen</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_Unmap">
       <property name="text">
        <string>entfernen</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_AutoMap">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Auto-Zuordnung</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_ClearMapping">
       <property name="text">
        <string>Alles löschen</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer_2">
       <property name="orientation">
        <enum>Qt::Orientation::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>57</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="grp_Options">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>90</y>
     <width>561</width>
     <height>181</height>
    </rect>
   </property>
   <property name="title">
    <string>Optionen</string>
   </property>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>321</x>
      <y>20</y>
      <width>231</width>
      <height>161</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="sizeConstraint">
      <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
     </property>
     <item>
      <widget class="QCheckBox" name="chb_BackupJSON">
       <property name="text">
        <string>Partrepository.json sichern</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="chb_HasHeader">
       <property name="text">
        <string>Erste Zeile enthält Spaltennamen</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="chb_StopOnError">
       <property name="text">
        <string>Bei Fehler abbrechen</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="chb_SkipEmptyRows">
       <property name="text">
        <string>Leere Zeilen überspringen</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="chb_TrimWhitespace">
       <property name="text">
        <string>Leerzeichen entfernen</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>14</x>
      <y>61</y>
      <width>293</width>
      <height>32</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="lbl_Delimiter">
       <property name="text">
        <string>Trennzeichen</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="cbb_Delimiter">
       <property name="editable">
        <bool>true</bool>
       </property>
       <property name="insertPolicy">
        <enum>QComboBox::InsertPolicy::NoInsert</enum>
       </property>
       <property name="sizeAdjustPolicy">
        <enum>QComboBox::SizeAdjustPolicy::AdjustToContents</enum>
       </property>
       <item>
        <property name="text">
         <string>;</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>,</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>\t</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>14</x>
      <y>93</y>
      <width>293</width>
      <height>23</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="lbl_SkipRows">
       <property name="text">
        <string>Zeilen überspringen</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spn_SkipRows">
       <property name="baseSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>14</x>
      <y>29</y>
      <width>293</width>
      <height>32</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="lbl_Encoding">
       <property name="text">
        <string>Encoding</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="cbb_Encoding">
       <item>
        <property name="text">
         <string>UTF-8</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>ISO-8859-1</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>14</x>
      <y>125</y>
      <width>293</width>
      <height>23</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="lbl_MaxRows">
       <property name="text">
        <string>Max. Zeilen importieren (0 = alle)</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spn_MaxRows">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="baseSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_4">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="grp_Preview">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>10</y>
     <width>451</width>
     <height>261</height>
    </rect>
   </property>
   <property name="title">
    <string>Vorschau</string>
   </property>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>431</width>
      <height>220</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_7">
     <item>
      <widget class="QTableView" name="tbl_Preview"/>
     </item>
     <item>
      <widget class="QLabel" name="lbl_PreviewInfo">
       <property name="text">
        <string>- Spalten, - Zeilen erkannt</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>560</y>
     <width>1021</width>
     <height>121</height>
    </rect>
   </property>
   <property name="title">
    <string>Log</string>
   </property>
   <widget class="QPlainTextEdit" name="txt_Log">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>1001</width>
      <height>81</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>21</x>
     <y>690</y>
     <width>1011</width>
     <height>32</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_6">
    <item>
     <widget class="QPushButton" name="btn_LoadPreview">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Vorschau laden</string>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="horizontalSpacer_5">
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>30</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QLabel" name="lbl_ImportStatus">
      <property name="minimumSize">
       <size>
        <width>100</width>
        <height>0</height>
       </size>
      </property>
      <property name="text">
       <string>Bereit...</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QProgressBar" name="prg_Import">
      <property name="value">
       <number>0</number>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="horizontalSpacer_6">
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QPushButton" name="btn_StartImport">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Importieren</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="btn_Close">
      <property name="focusPolicy">
       <enum>Qt::FocusPolicy::StrongFocus</enum>
      </property>
      <property name="text">
       <string>Abbrechen</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>edt_ImportFileName</tabstop>
  <tabstop>btn_BrowseCsv</tabstop>
  <tabstop>cbb_Encoding</tabstop>
  <tabstop>cbb_Delimiter</tabstop>
  <tabstop>spn_SkipRows</tabstop>
  <tabstop>chb_BackupJSON</tabstop>
  <tabstop>chb_HasHeader</tabstop>
  <tabstop>chb_SkipEmptyRows</tabstop>
  <tabstop>chb_StopOnError</tabstop>
  <tabstop>chb_TrimWhitespace</tabstop>
  <tabstop>spn_MaxRows</tabstop>
  <tabstop>lst_CsvColumns</tabstop>
  <tabstop>btn_MapToField</tabstop>
  <tabstop>btn_Unmap</tabstop>
  <tabstop>btn_AutoMap</tabstop>
  <tabstop>btn_ClearMapping</tabstop>
  <tabstop>lst_PartFields</tabstop>
  <tabstop>tbl_Mapping</tabstop>
  <tabstop>txt_Log</tabstop>
  <tabstop>btn_LoadPreview</tabstop>
  <tabstop>tbl_Preview</tabstop>
  <tabstop>btn_StartImport</tabstop>
  <tabstop>btn_Close</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
